player/loadfile.c

```diff
@@ -239,8 +239,11 @@ static void uninit_demuxer(struct MPContext *mpctx)
 #define WHITE_CIRCLE "\xe2\x97\x8b"
 #define APPEND(s, ...) mp_snprintf_cat(s, sizeof(s), __VA_ARGS__)
 #define FILL(s, n) mp_snprintf_cat(s, sizeof(s), "%*s", n, "")
+#define ADD_FLAG(b, flag, first) \
+    APPEND(b, " %s%s", first ? "[" : "", flag); \
+    first = false;
```

Comment:
Ok, last one.

Just small suggestion, wrap it in do while, so you don't accidentally forget about `{}`. And if you do that, you can remove `{}` from ifs below.
```suggestion
#define ADD_FLAG(b, flag, first) do {            \
    APPEND(b, " %s%s", first ? "[" : "", flag);  \
    first = false;                               \
} while(0)
```